### ISSUE TITLE
Nodes with version mismatch should be switched to correct version before adding to the cluster

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#872]
Originally reported of Google Code with ID 816.

```
<b>What steps will reproduce the problem?</b>
1. Cluster running Ganeti 2.10
2. One new node that has both Ganeti 2.10 and 2.11 available where the symlinks point to 2.10
3. Try to add that node to the cluster
4. Getting  a "Version mismatch"

<b>What is the expected output? What do you see instead?</b>
The symlinks on the new node get switched to 2.10 and the node gets added

Probably the easiest way would be to check the target of the symlink on the master node and try to switch the new node to the same version.
```
Originally added on 2014-04-28 12:51:25 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-04-28 12:51:56 +0000 UTC.

Changed State: New

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-04-28 12:56:23 +0000 UTC.

Added Labels: Priority-High
Changed State: Accepted
Added to Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit edc6dbc6a9c5473ddde45018561caf64be209a5d
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Apr 29 11:13:42 2014 +0200

    Set correct Ganeti version on setup commands
    
    When asked to execute a setup command, prefix it with
    a command sequence to test for the existence of the
    needed Ganeti version and switching to it.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit a5bd9ea7090ece14ee3dc660963745835a1a369d
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Apr 28 18:20:45 2014 +0200

    Add a utility to combine shell commands
    
    Add a function that combines individual shell commands into
    a single command (calling the standard shell) that executes
    the given commands in sequence while they succeed.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2014-04-29 10:48:02 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The mentioned patches are merged up to 2.11 as well.
```
Originally added on 2014-04-29 11:11:44 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
2.11.6 released
```
Originally added on 2014-09-22 15:56:02 +0000 UTC.

Changed State: Released
